lib/_stream_readable.js

```diff
@@ -211,7 +211,7 @@ function howMuchToRead(n, state) {
   if (state.objectMode)
     return n === 0 ? 0 : 1;
 
-  if (n === null || isNaN(n)) {
+  if (n === null || n !== n) {
```

Comment:
Might be a little paranoid here and totally not related to this PR, but If `n` is not a number, then we might fail somewhere with improper error message I guess.